components/gitlab/new-milestone.js

```diff
@@ -0,0 +1,91 @@
+const gitlab = require("https://github.com/PipedreamHQ/pipedream/components/gitlab/gitlab.app.js");
+
+module.exports = {
+  name: "New Milestone (Instant)",
+  description: "Triggers when a new milestone is created in a project",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    gitlab,
+    projectId: { propDefinition: [gitlab, "projectId"] },
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60,
+      },
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
+      const opts = {
+        projectId: this.projectId,
+        pageSize: 1,
+      };
+      const milestones = await this.collectMilestones(opts);
+      if (milestones.length > 0) {
+        const lastProcessedMilestoneId = milestones[0].id
+        this.db.set("lastProcessedMilestoneId", lastProcessedMilestoneId);
+        console.log(
+          `Polling Gitlab milestones created after ID ${lastProcessedMilestoneId}`
+        );
+      }
+    },
+  },
+  methods: {
+    generateMeta(data) {
+      const {
+        id,
+        created_at,
+        title,
+      } = data;
+      const summary = `New milestone created: ${title}`;
```

Comment:
Since there's limited visible space in the event summary I normally like to emit just the name of the relevant thing (in this case just the `title`, since the fact that it's a new milestone should be obvious from context.

Reply:
Done